### PR TITLE
log: Improve Error: backup lock exists 

### PR DESCRIPTION
**Pull request by zwj-coder**
### What problem does this PR solve? 
When user want to backup data in a directory that has stored the backup data from previous backup, and meta or lock file exists error will occur, it's better to notice user the path of the existed file.

### What is changed and how it works?
Improve error message in `pkg/client.go`, and add unit test

The previous error is as follow:
```
Error: backup lock exists, may be some backup files in the path already: [BR:Common:ErrInvalidArgument]invalid argument
```
Now the error is as follow:
```
Error: backup meta file exists in local:///tmp/backup_test/backupmeta, there may be some backup files in the path already, please specify a correct backup directory!: [BR:Common:ErrInvalidArgument]invalid argument
```

### Check List 

Tests 

 - Unit test
 - Manual test

Code changes

 - Has exported function/method change

Related changes

 - Need to cherry-pick to the release 5.0 and release 4.0

### Release note

 - No Release Note

### CODE REVIEW

**Review by YuJuncen — APPROVED** [review]
lgtm

**Comment by IANTHEREAL** [comment]
Good Job!

How about putting `[BR:Common:ErrInvalidArgument]invalid argument` in front of `Error: backup lock file exists in local:///tmp/backup_test/backupmeta, may be some backup files in the path already`
and I suggest add a workaround, like 
```
[BR:Common:ErrInvalidArgument]invalid argument: backup lock file exists in local:///tmp/backup_test/backupmeta, may be some backup files in the path already, please specify a correct backup directory!
```

**Comment by kennytm** [comment]
Looks like a unit test failed without message

```
[2021-06-15T03:00:59.896Z] OOPS: 63 passed, 1 FAILED

[2021-06-15T03:00:59.896Z] --- FAIL: TestTiDB (30.27s)
```

File: pkg/backup/client.go [hunk 1]
```diff
@@ -176,14 +193,20 @@ func (bc *Client) SetStorage(ctx context.Context, backend *backuppb.StorageBacke
 		return errors.Annotatef(err, "error occurred when checking %s file", metautil.MetaFile)
 	}
 	if exist {
-		return errors.Annotate(berrors.ErrInvalidArgument, "backup meta exists, may be some backup files in the path already")
+		errMsg := fmt.Sprintf(`backup meta file exists in %v, 
+							there may be some backup files in the path already, 
+							please specify a correct backup directory!`, GetLockOrMetaFilePath(metautil.MetaFile, backend))
```
Comment (kennytm) [inline pkg/backup/client.go:198]: `GetLockOrMetaFilePath(metautil.MetaFile, backend)` should be equivalent to `bc.storage.URI() + "/" + metautil.MetaFile`.

Comment (zwj-coder) [reply pkg/backup/client.go]: removed redundant code

**Review by kennytm — APPROVED** [review]

**Comment by IANTHEREAL** [comment]
/lgtm 

Need to adjust the log output order, `[BR:Common:ErrInvalidArgument] invalid argument` should be better to be placed in front of other messages. I think we can file an issue to `errors` repo -  the output order is very unreasonable

**Review by 3pointer — APPROVED** [review]

**Pull request merged**